PhysicsTools/NanoAOD/plugins/TriggerOutputBranches.cc

```diff
@@ -21,7 +21,8 @@ void TriggerOutputBranches::updateTriggerNames(TTree& tree,
     for (unsigned int j = 0; j < newNames.size(); j++) {
       std::string name = newNames[j];  // no const & as it will be modified below!
       std::size_t vfound = name.rfind("_v");
-      if (vfound != std::string::npos && (name.compare(0, 3, "HLT") == 0 || name.compare(0, 2, "L1") == 0)) {
+      if (vfound != std::string::npos && ((name.compare(0, 3, "HLT") == 0) || name.compare(0, 2, "L1") == 0 ||
+                                          (m_isPFScouting && name.find("PFScouting") != std::string::npos))) {
```

Comment:
can you please find a way to not have to use and specify `m_isPFScouting` ? what is the naming convention that prevents from just checking on the composition of `name` ?

Reply:
There are two paths starting with DST (`DST_HLTMuon_Run3_PFScoutingPixelTraciking` and `DST_Run3_PFScoutingPixelTracking`) and one with MC (`MC_Run3_PFScoutingPixelTracking`) which would be useful to store for scouting. To avoid storing them also in the standard NanoAOD, I added the `m_isPFScouting` variable. But I understand your point. However I'm not sure how to implement it so that it is stored in the scouting Nano and not the standard Nano without this variable.

Reply:
if `m_isPFScouting` is only need to not store extra path in standard nano, then you can get rid if it and use only `name.find("PFScouting")`. Though I think, that the naming of these trigger path should be rationalised to `Scouting_...` (or anything else that @cms-sw/hlt-l2 would suggest)

Reply:
> (or anything else that https://github.com/orgs/cms-sw/teams/hlt-l2 would suggest)

Since we were mentioned, I'll just give my 2 cents (with the caveat that I didn't look at this PR in detail, I know little about nanoAOD, and I don't have to sign this PR): I would rely as little as possible on the names of the HLT paths, and when those ultimately have to be specified, they should be specified in the python configuration (not in the cpp implementation) with wildcards/regexs. Along these lines, I agree that `m_isPFScouting` should not be part of the cpp implementation. Finally, while `PFScouting` is the correct label for 2022/23, `(S|s)couting` sounds a bit more future-proof (and could even work for Run 2). I am, anyway, missing something about this PR, because I don't see the parameter "isPFScouting" specified anywhere in the python files (shouldn't it be `true` for the new 'ScoutingNano' wf?).

Reply:
Thanks for your comments! Taking them into consideration, I think I have an idea of how this could be implemented better with vector of HLT paths which is read from the python configuration. I will update the PR shortly.